express-post-json/server.ts

```diff
@@ -0,0 +1,37 @@
+import express from 'express';
+
+type Student = {
+  id?: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type Students = Record<number, Student>;
+
+let nextID: number = 1;
+const grades: Students = {};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradesResponse: Student[] = [];
+  for (const key in grades) {
+    gradesResponse.push(grades[key]);
+  }
+  res.json(gradesResponse);
+});
+
+const jsonParser = express.json();
+app.use(jsonParser);
+
+app.post('/api/grades', (req, res) => {
+  grades[nextID] = req.body;
```

Comment:
Using the types in my comment above, you could write this line to provide type checking like this:
```
const newStudent: UnsavedStudent = req.body;
grades[nextID] = { id: nextID, ...newStudent };
```
Note that you need to make sure the `id` exists in the object you are assigning to `grades[nextID]` since that is declared as `Student`.